src/interfaces.ts

```diff
@@ -66,6 +66,10 @@ export interface IFieldConfigOptionSelect extends IFieldConfigBase {
   sorted?: boolean;
 }
 
+export interface IFieldConfigAddress extends IFieldConfigBase {
+  stateProps: IFieldConfigOptionSelect;
```

Comment:
Since optionSelect is a full fieldConfig, and this does not require `field` ( plus field makes no sense here ), perhaps it should be Partial<IFieldConfigOptionSelect>? Also since this won't be filled in in the `fillInFieldConfig` it's actually `stateProps?:` right?